File: src/KubeOps.Templates/Templates/Operator.CSharp/Finalizer/DemoFinalizer.cs
using k8s.Models;

using KubeOps.Abstractions.Finalizer;

using Microsoft.Extensions.Logging;

using GeneratedOperatorProject.Entities;

namespace GeneratedOperatorProject.Finalizer;

public class DemoFinalizer(ILogger<DemoFinalizer> logger) : IEntityFinalizer<V1DemoEntity>
{
    public Task FinalizeAsync(V1DemoEntity entity, CancellationToken cancellationToken)
    {
        logger.LogInformation($"entity {entity.Name()} called {nameof(FinalizeAsync)}.");

        return Task.CompletedTask;
    }
}

File: _old/src/KubeOps.Templates/Templates/Operator.FSharp/Finalizer/DemoFinalizer.fs
namespace GeneratedOperatorProject.Finalizer

open System.Threading.Tasks
open GeneratedOperatorProject.Entities
open k8s.Models
open KubeOps.Operator.Finalizer
open Microsoft.Extensions.Logging

type DemoFinalizer(logger: ILogger<DemoFinalizer>) =
    interface IResourceFinalizer<V1DemoEntity> with
        member this.FinalizeAsync entity =
            logger.LogInformation $"entity {entity.Name()} called finalize"
            Task.CompletedTask
